pkg/ibmvpc/vpc.go

```diff
@@ -713,6 +742,12 @@ type TransitGateway struct {
 	destNodes   []vpcmodel.Node
 
 	region *Region
+
+	// maps APs of each VPC to the prefix that determines its connectivity status w.r.t. the tgw (allow/deny)
+	// if non default. Specifically, the map is from VPC UID to a map between the ap's ipBlock to the index of the matching prefix
+	// that determines its status;
+	// this struct can be though of as the "explain" parallel of availableRoutes; note that it also lists deny prefixes
+	vpcApsPrefixes map[string]map[*ipblock.IPBlock]tgwPrefix
```

Comment:
Actually this map is not used as a map. Only the value of the `*ipblock.IPBlock` is used. 
Changing the map to used string and performing the required conversions seems to much work given that we don't actually need a map.
We can either add a comment regarding this issue, or change the struct used to avoid future confusion. In the latter case we will need to define a dedicated struct to that end, and use a slice of this newly defined struct. Any other idea? @adisos

Reply:
done

Reply:
I think the documentation for this field it still outdated? also can it be made more simple/concise ?

Reply:
possible changes:
```
	// maps each VPC UID to a slice - listing for each of its AP (as IPBlock) details of the matching filter
       // Specifically, these details include the specific transit connection and the index of the matching
	// filter if exists (index "-1" is for default )
	// this struct can be thought of as the "explain" parallel of availableRoutes; note that unlike availableRoutes it also lists deny prefixes
```

Reply:
No, its not supposed to be. Here, this is what you should have seen (was already pushed). Is this what you referred to?
```

	// maps each VPC to a struct listing for each of its address prefix details of the filter prefix determining its behavior w.r.t.
	// the tgw. Specifically, these details include the specific transit connection and the index of the matching
	// (allow/deny) filter prefix rule if exists, or -1 if the behavior is determined by the specified default of the transit connection
	// this struct can be though of as the "explain" parallel of availableRoutes; note that unlike availableRoutes it also lists deny prefixes
	vpcsAPToFilters map[string][]IPBlockPrefixFilter
```